src/chttpd/test/exunit/pagination_test.exs

```diff
@@ -1163,7 +1221,153 @@ defmodule Couch.Test.Pagination do
             assert length(result["rows"]) > ctx.page_size
           end)
         end
+
+        test "can retrieve all pages", ctx do
+          [descending_query, limit_query] =
+            paginate_queries(
+              ctx,
+              url:
+                "/#{ctx.db_name}/_design/#{ctx.ddoc_id}/_view/#{ctx.view_name}/queries",
+              query: %{page_size: ctx.page_size, descending: ctx.descending},
+              body: :jiffy.encode(ctx.queries)
+            )
+
+          results = List.flatten(descending_query)
+          assert ctx.n_docs == length(results)
+          expected_key_order = :descending
+          expected_ids_order = :ascending
+          assert expected_key_order == ordering?(results, "key"),
+            "expecing keys in #{expected_key_order} order, got: #{inspect field(results, "key")}"
+          assert expected_ids_order == ordering?(results, "id"),
+            "expecing ids in #{expected_ids_order} order, got: #{inspect field(results, "id")}"
+
+          results = List.flatten(limit_query)
+          expected_length = if ctx.page_size + 2 > ctx.n_docs do
+            ctx.page_size + 1 - 2
+          else
+            ctx.page_size + 1
+          end
```

Comment:
How does the calculation work out here, where do 1 and 2 come from?

Reply:
I'll change the calculation to be easier to understand:
```
[_descendiing_query, query] = ctx.queries[:queries]
expected_length =
      if ctx.n_docs - query.skip > query.limit do
          query.limit
      else
          query.limit - query.skip
      end
```